docs: Update Memory Tutorial Example (#1242)
block/goose: documentation/docs/tutorials/memory-mcp.md

import YouTubeShortEmbed from '@site/src/components/YouTubeShortEmbed';

> The Memory extension turns Goose into a knowledgeable assistant by allowing you to teach it **personalized key information**—like commands, code snippets, browser preferences and configurations—that it can recall and apply later. Whether it’s project-specific (local) or universal (global) knowledge, Goose learns and remembers what matters most to you.

Reviewer: ```suggestion
The Memory extension turns Goose into a knowledgeable assistant by allowing you to teach it **personalized key information** (e.g. commands, code snippets, preferences and configurations) that it can recall and apply later. Whether it’s project-specific (local) or universal (global) knowledge, Goose learns and remembers what matters most to you.
```

> This tutorial will cover enabling and using the Memory MCP Server, which is a built-in Goose extension. 

Reviewer: ```suggestion
This tutorial covers enabling and using the Memory MCP Server, which is a built-in Goose extension. 
```

  </TabItem>
</Tabs>

## Why Use Memory?  

> With the Memory extension, you’re not just storing static notes—you’re teaching Goose how to assist you better. Imagine telling Goose:  

Reviewer: ```suggestion
With the Memory extension, you’re not just storing static notes, you’re teaching Goose how to assist you better. Imagine telling Goose:  
```

> - **"Goose, learn everything about MCP servers and save it."**  

Reviewer: ```suggestion
- **"Goose, learn everything about MCP servers and save it to memory."**  
```

- Later, you can ask: **"Goose, utilizing our MCP server knowledge help me build an MCP server."**  

> Goose will recall everything you’ve saved—your preferences, past attempts, what worked, and what didn’t—making it easier to pick up where you left off and build consistently.

Reviewer: Goose wont remember past attempts, what worked and what didn't unless you told it to remember. Maybe add that above to the prompt?

Author: that line I was expanding on what goose can remember, that's why i said:
it'll recall everything you've saved ...

> Goose will recall everything you’ve saved, such as your preferences, past attempts, what worked, and what didn’t, as long as you instruct it to remember. This makes it easier to pick up where you left off and build consistently.

Reviewer: ```suggestion
Goose will recall everything you’ve saved as long as you instruct it to remember. This makes it easier to have consistent results when working with Goose.
```

## Example Usage

In this example, I’ll show you how to make Goose a knowledgeable development assistant by teaching it about your project’s API standards. With the Memory extension, Goose can store structured information and retrieve it later to help with your tasks.

> This means you no longer have to repeat yourself—Goose will remember your project’s requirements and automatically apply them to new tasks.

Reviewer: ```suggestion
This means you no longer have to repeat yourself. Goose will remember your project’s requirements and automatically apply them to new tasks.
```

  </TabItem>
</Tabs>

### Step 1: Teach Goose Your API Standards

>     #### Goose Prompt #1

Reviewer: great one!
